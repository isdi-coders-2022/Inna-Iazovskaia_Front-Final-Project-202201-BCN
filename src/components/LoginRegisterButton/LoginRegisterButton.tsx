import styled from "styled-components";
import { ButtonInterface } from "../../types/Interfaces";

const Button = styled.button`
  border-radius: 20px;
  border-color: rgba(156, 166, 255, 1);
  color: white;
  height: 45px;
  font-size: 18px;
  width: 100%;
`;

const LoginRegisterButton = ({
  actionOnClick,
  text,
  className,
  disabled,
}: ButtonInterface): JSX.Element => {
  return (
    <Button className={className} onClick={actionOnClick} disabled={disabled}>
      {text}
    </Button>
  );
};

export default LoginRegisterButton;
