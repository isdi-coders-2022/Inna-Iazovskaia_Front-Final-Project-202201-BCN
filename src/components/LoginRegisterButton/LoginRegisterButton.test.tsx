import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import LoginRegisterButton from "./LoginRegisterButton";
import TestRenderer from "react-test-renderer";

describe("Given a LoginRegisterButton copmponent", () => {
  describe("When it receives 'Push' as its text", () => {
    test("Then it should render button with text 'Push'", () => {
      const text = "Push";
      const className = "";

      render(<LoginRegisterButton text={text} className={className} />);

      const button = screen.getByRole("button", { name: text });

      expect(button).toBeInTheDocument();
    });
  });

  describe("When it receives 'button-class' as its className", () => {
    test("Then it should render button with class 'button-cless'", () => {
      const text = "";
      const className = "button-cless";

      render(<LoginRegisterButton text={text} className={className} />);

      const button = screen.getByRole("button");

      expect(button).toHaveClass(className);
    });
  });

  describe("When it receives an action and a click", () => {
    test("Then it should execute the action", () => {
      const actionOnClick = jest.fn();
      const text = "";
      const className = "";

      render(
        <LoginRegisterButton
          text={text}
          className={className}
          actionOnClick={actionOnClick}
        />
      );

      userEvent.click(screen.getByRole("button"));

      expect(actionOnClick).toHaveBeenCalled();
    });
  });

  describe("When it's rendered'", () => {
    test("Then it should always match this snapshot", () => {
      const text = "";
      const className = "";
      const renderedButton = TestRenderer.create(
        <LoginRegisterButton text={text} className={className} />
      );
      expect(renderedButton).toMatchSnapshot();
    });
  });
});
