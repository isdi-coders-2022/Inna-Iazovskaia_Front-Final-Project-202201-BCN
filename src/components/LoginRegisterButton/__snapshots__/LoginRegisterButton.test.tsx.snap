// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a LoginRegisterButton copmponent When it's rendered' Then it should always match this snapshot 1`] = `
<button
  className="sc-bdvvtL bolGWz"
>
  
</button>
`;
